frontend/src/stores/knowledges/getters.ts

```diff
@@ -1,8 +1,23 @@
 import { computed } from 'vue'
 import { knowledges } from '@/stores/knowledges/state'
+import type { Role } from '@/domain/Role'
+import type { Knowledge } from '@/domain/Knowledge'
+import { get } from '@/services/http'
 
 const getKnowledge = computed(() => {
   return knowledges?.value
 })
 
-export { getKnowledge }
+async function getKnowledgesByRoleId(roleId: number): Promise<Knowledge[]> {
+  if (knowledges?.value?.length > 0) {
+    return knowledges.value
+  }
+  const roleResponse = await get<Role>(`/roles/${roleId}`)
+
+  if (!roleResponse) {
+    return []
+  }
+  return roleResponse.knowledges
```

Comment:
```suggestion
  return roleResponse?.knowledges
```

Doing this, you don't need the previous if. This means "If roleResponse exists, return its knowledges"